src/init/chainstate.cpp

```diff
@@ -21,6 +20,7 @@ std::optional<ChainstateLoadingError> LoadChainstateSequence(bool fReset,
                                                              unsigned int check_blocks,
                                                              unsigned int check_level,
                                                              std::function<int64_t()> get_unix_time_seconds,
+                                                             std::optional<std::function<bool()>> shutdown_requested,
```

Comment:
In commit "init/chainstate: Decouple from ShutdownRequested"

Without actually trying it.. I think it would be more straightforward here to use a `CThreadInterrupt` which gets triggered for this.